addons/mail/controllers/thread.py

```diff
@@ -131,7 +131,8 @@ def mail_message_post(self, thread_model, thread_id, post_data, context=None, **
                 for key, value in post_data.items()
                 if key in thread._get_allowed_message_post_params()
             }
-        message = thread.message_post(
+        # sudo: mail.thread - can post message on accessible thread
+        message = thread.sudo().message_post(
```

Comment:
Note: this sudo only acceptable after rebasing on https://github.com/odoo/odoo/pull/179908